mysite/code/tasks/SetSupportedAddonsTask.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Provides a CLI for bulk setting supported addons
+ *
+ * @package mysite
+ */
+class SetSupportedAddonsTask extends BuildTask
+{
+    /**
+     * {@inheritDoc}
+     * @var string
+     */
+    protected $title = 'Set Supported Addons';
+
+    /**
+     * {@inheritDoc}
+     * @var string
+     */
+    protected $description = 'Bulk update addons that should be registered as supported by SilverStripe';
+
+    /**
+     * {@inheritDoc}
+     * @param SS_HTTPRequest $request
+     */
+    public function run($request)
+    {
+        $vars = $request->getVars();
+
+        // Check for args
+        $args = $this->processArgs($request->getVar('args') ?: []);
+        $isReplace = in_array('r', $args);
+        $isForce = in_array('f', $args);
+
+        unset($vars['url']);
+        if (empty($vars) && !$isReplace) {
+            $this->outputHelp();
+            return;
+        }
+
+        $addons = $request->getVar('addons');
+
+        if ($isReplace) {
+            // Catch a mistake of replacing addons but providing no addons to replace it with
+            if (empty($addons) && !$isForce) {
+                $this->output('Providing no addons when using -r will unsupport all addons. Use -f to force.');
+                return;
+            }
+
+            // Mark all addons as unsupported
+            $this->output('All addons have been marked unsupported');
+            SQLUpdate::create('Addon', ['Supported' => false])->execute();
+
+            // If there are no addons to replace with, say as such and return
+            if (empty($addons)) {
+                return;
+            }
+        }
+
+        $addons = explode(',', $addons);
+        $addonList = Addon::get()->filter('Name', $addons);
+
+        foreach ($addons as $addon) {
+            if ($addonObject = $addonList->find('Name', $addon)) {
+                $addonObject->Supported = true;
+                $addonObject->write();
+                $this->output($addon . ' marked as supported');
+            } else {
+                $this->output($addon . ' did not match a known addon');
+            }
+        }
+    }
+
+    /**
+     * @param string $output
+     */
+    protected function output($output)
+    {
+        echo $output . PHP_EOL;
+    }
+
+    protected function outputHelp()
+    {
+        $className = __CLASS__;
+
+        $this->output(<<<HELP
+{$this->description}
+
+Usage: framework/sake dev/tasks/$className [-fr] addons=foo/bar,bin/baz
+
+ -r Replace existing supported addons with the new given list
+ -f Used with -r and an empty set of addons to drop support for all addons
+ 
+HELP
+        );
+    }
+
+    /**
+     * @param array $args
+     * @return array
+     */
+    protected function processArgs(array $args)
+    {
+        $outputArgs = [];
+
+        foreach ($args as $arg) {
+            $outputArgs = array_merge($outputArgs, str_split(substr($arg, 1)));
+        }
+
+        return $outputArgs;
+    }
+}
```

Comment:
I actually disagree with having a build task for this - we can use the admin interface to set addons as supported manually. What do you think?

Reply:
I thought it wouldn't hurt. I can't imagine it would change much, but when it does change it feels cumbersome weeding through the pages of addons to tick some boxes - especially if you're looking for specific addons.

Reply:
You can always search 😄 what about the longer term considerations, for example if we _were_ going to add a support semver constraint then how would that fit into this build task? The admin interface provides searching, exporting and importing to CSV out of the box - we'd just need to make sure that this task justifies its existence enough

Reply:
That's a good point. It was mostly an exercise to learn about tasks. I can remove it :)